### PR TITLE
Prove DFlash and EAGLE3 GGUF draft runtime

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- Add exact target/draft GGUF pair construction with separate cache namespaces, target hidden-state outputs, shared embedding/LM-head bridges, atomic package publication, and self-verifying graph identities.
- Add a direct ONNX Runtime greedy coordinator for DFlash and EAGLE3 with accepted-prefix verification, target/draft cache rollback, compressed-vocabulary remapping, strict output-capacity bounds, and target-only baseline comparison.
- Add a separate test-only direct ORT oracle that imports neither `DraftPairRunner` nor production transition helpers, derives target-layer/remap/block/mask contracts from raw immutable GGUF metadata, and compares production behavior against independently recorded transitions.
- Make EAGLE3 consume real draft-owned embeddings, normalize GGUF/HuggingFace hidden-state indices correctly, and preserve fully-owned cross-width variants.
- Semantically rebase onto exact current `main` at `d68f66418096e0fe9a073764b819d6898de314de`, integrating #696, #693, and #691 while retaining shared MTP cache/publication/status infrastructure without duplication.
- Keep higher-level runtime compatibility advisory; no production revision pin or downstream runtime gate is added.

## Immutable evidence

Evidence is bound in `testdata/evidence/gguf_draft_runtime_evidence.json` (13,293,887,389 bytes total, below the 16 GiB cap), including artifact/config/tokenizer/source/graph/package/runtime hashes. Separate SHA-256-bound independent traces record every proposal ID/remapped token and proposal-logit digest, posterior/accepted prefix/correction, target verification/replay logits and tokens, target/draft cache lengths and tensor digests before/tentative/committed/replay states, final cache states, exact counters, and the explicit batch-size-one reorder disposition.

- **DFlash:** `Qwen/Qwen3-4B-GGUF@bc640142...` + `AtomicChat/Qwen3-4B-DFlash-GGUF@b53c7de6...`; 32/32 direct-ORT greedy tokens equal target-only, 18/75 proposals accepted, 3 multi-token rounds, 5 rejected-suffix rollbacks. Independent full-tensor Q8 source comparison: cosine `0.999879`, relative L2 `0.005406`.
- **EAGLE3:** `Qwen/Qwen3-8B-GGUF@7c41481f...` + `williamliao/Qwen3-8B-EAGLE3-Speculator-GGUF@44480ff4...`; 32/32 direct-ORT greedy tokens equal target-only, 20/43 proposals accepted, 7 multi-token rounds, 10 rejected-suffix rollbacks. Independent pinned llama.cpp conversion matches all 15 tensor names, qtypes, shapes, and values exactly.
- Pinned llama.cpp additionally exercised the real acceptance loops with 49/114 DFlash and 45/85 EAGLE3 accepted tokens.
- Four mutations now alter execution before validation: wrong mapping/order is fed through target verification, draft-cache corruption is threaded into the next round, and target rollback corruption is replayed against full committed history. Each fails its distinct semantic invariant.

Target Q4_K/Q6_K imports truthfully report lossy INT4 affine block-32 repacking; target-only and speculative comparisons use the same repacked target graph. Every target replay starts from an empty cache and rebuilds the complete prompt plus committed token history. Quantized batched verification and full-history replay can differ numerically, so both states are bound and maximum absolute differences are recorded (DFlash `1.967625`, EAGLE3 `0.912386`) while replay argmax tokens must match exactly. Draft replay likewise reconstructs the complete committed history from empty state.

Production DFlash uses target-only tail generation when fewer than one fixed block remains; EAGLE3 caps each final proposal width to `remaining - 1`. Parametric tests prove target sessions, caches, and counters never process beyond requested `max_new_tokens` for remaining capacity 1 through width+1. The reference coordinator is batch-size 1 and truthfully reports beam/reorder unsupported. Higher-level execution remains `runtime_unvalidated` without gating export.

## Validation

- Final exact-main broad non-integration suite: 9,668 passed, 64 skipped, 1 subtest passed.
- Final affected draft+MTP+CLI/evidence/docs/census suite: 302 passed, 2 real-artifact probes skipped by opt-in.
- Both immutable real-artifact probes were rerun after the three blocker fixes; all eight execution-mutating discriminator runs failed at their expected semantic invariants.
- Pinned Ruff format/lint, generated GGUF docs, and focused isolated mypy passed. `_config_mapping.py` retains three unrelated pre-existing mypy errors outside the DFlash change.
- Final medium review specifically verified execution-mutating discriminators, complete empty-cache target replay, bounded final-round production behavior, DFlash mask metadata propagation, and semantic #693/#691 conflict resolution; no significant findings remain.

No CI monitoring or merge is requested.

### CODE REVIEW

File: src/mobius/integrations/gguf/_docs.py [hunk 1]
```diff
@@ -663,16 +704,11 @@ def render_document() -> str:
 package.save("output")
 ```
 
-`keep_quantized=True` requests quantized target storage where supported; it does
-not guarantee source byte or numerical fidelity. Mobius classifies qtypes before
-payload conversion, emits one aggregate warning for lossy requantization, and
-saves the typed result as `quantization_report.json`. Use
-`keep_quantized=False` for explicit float import.
+For target-coupled DFlash/EAGLE3, use `build_draft_pair_from_gguf`, `write_draft_pair_package`, and `DraftPairRunner`; the package carries independent caches and required target bridges, while `runtime_unvalidated` warns about higher-level runtimes without gating direct ORT. CLI: `mobius build-gguf draft.gguf --target-gguf target.gguf --target-config target-config --output output`.
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/gguf/_docs.py]: Fixed by constructing the generated paragraph from wrapped source fragments. The rendered document stays on one line for the generated-doc concision contract, and the pinned Ruff checks pass.

File: src/mobius/integrations/gguf/_draft_runtime.py [hunk 2]
```diff
@@ -0,0 +1,738 @@
...
+        pending_features = target_features[:, -1:]
+        remap = self.manifest.get("draft_to_target")
+        if remap is not None and (not isinstance(remap, list) or not remap):
+            raise ValueError("EAGLE3 draft_to_target must be null or a non-empty map")
+        rounds: list[tuple[int, int]] = []
+        rollbacks: list[tuple[int, int, int, int]] = []
+
+        while len(generated) < max_new_tokens:
+            start = input_ids.shape[1] + len(generated) - 1
+            base_cache = {name: value.copy() for name, value in draft_cache.items()}
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/gguf/_draft_runtime.py]: Fixed. The accepted draft cache dictionary is now reused directly as the rollback baseline; ORT receives immutable input arrays and each tentative step returns new `present.*` arrays.
